swatch-core/src/main/java/org/candlepin/subscriptions/clowder/ClowderJsonPathPropertySource.java

```diff
@@ -273,4 +376,46 @@ private String findPropertySourceInsertionPoint(MutablePropertySources sources)
     }
     return StandardEnvironment.SYSTEM_PROPERTIES_PROPERTY_SOURCE_NAME;
   }
+
+  @SuppressWarnings("unchecked")
+  private Object getKafkaBrokerConfig(String name, KafkaBrokerConfigMapper configFunction) {
+    Object value = getJsonPathValue(KAFKA_BROKERS);
+    if (value instanceof Collection<?> list) {
+      if (list.isEmpty()) {
+        return null;
+      }
+
+      List<Map<String, Object>> brokers = list.stream().map(o -> (Map<String, Object>) o).toList();
+      return configFunction.apply(brokers);
+    } else {
+      throw new IllegalStateException(
+          "Unknown type found in clowder configuration for property: " + name);
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private static String getKafkaBrokerSaslProperty(
+      List<Map<String, Object>> brokerConfig, String propertyName) {
+    Map<String, Object> broker = getKafkaBrokerWithSasl(brokerConfig);
+    if (broker != null) {
+      Map<String, Object> saslConfig = (Map<String, Object>) broker.get("sasl");
+      Object value = saslConfig.get(propertyName);
+      if (value instanceof String str) {
+        return str;
+      }
+    }
+
+    return null;
+  }
+
+  private static Map<String, Object> getKafkaBrokerWithSasl(
+      List<Map<String, Object>> brokerConfig) {
+    return brokerConfig.stream()
+        .filter(m -> "sasl".equals(m.get("authtype")))
+        .findFirst()
```

Comment:
I'd like to be a little stricter here.  I.e. throw an exception if there is more than one SASL configuration given.  I want to fail explicitly and fast if we get JSON like

```json
           {
                "hostname": "env-rhsm-kafka.rhsm.svc",
                "port": 29092
                "port": 29092,
                "cacert": "Dummy value",
                "authtype": "sasl",
                "sasl": {
                    "username": "john",
                    "password": "doe",
                    "securityProtocol":"SASL_SSL",
                    "saslMechanism":"PLAIN"
                }
            },
           {
                "hostname": "env-rhsm-kafka-2.rhsm.svc",
                "port": 29092
                "port": 29092,
                "cacert": "A different cert",
                "authtype": "sasl",
                "sasl": {
                    "username": "jane",
                    "password": "smith",
                    "securityProtocol":"SASL_SSL",
                    "saslMechanism":"PLAIN"
                }
            }
```

There's a Guava collector [`toOptional`](https://javadoc.scijava.org/Guava/com/google/common/collect/MoreCollectors.html#toOptional()) that is useful for this purpose.

Reply:
I don't follow what is the value that the Guava collector toOptional brings here (we could simply don't use the `orElse`). Also, I think the code looks more readable without using optionals, and the methods are private and meant to be used internally only). 

About being stricter here, this is a change of behavior since previously, this is how it was working and also it's how it works the Clowder config source for Quarkus. We can always do this change in a later PR if there is a need.